src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTask.java

```diff
@@ -144,6 +149,31 @@ public void stop() {
     }
   }
 
+  private Consumer<MongoProcessedSinkRecordData> createErrorReporter() {
+    Consumer<MongoProcessedSinkRecordData> errorReporter = processedSinkRecordData -> {};
+    if (context != null) {
+      try {
```

Comment:
This logic is per: https://kafka.apache.org/26/javadoc/org/apache/kafka/connect/sink/SinkTaskContext.html#errantRecordReporter--